app/vmalert/main.go

```diff
@@ -299,12 +299,9 @@ func getAlertURLGenerator(externalURL *url.URL, externalAlertSource string, vali
 		"tpl": externalAlertSource,
 	}
 	return func(alert notifier.Alert) string {
-		qFn := func(_ string) ([]datasource.Metric, error) {
```

Comment:
why this change was made? the error message here looks more precise and clear to user than internal message

Reply:
I don't see the need to define an invalid qFn() everywhere, as the error will be wrapped in the following line:
```
return nil, fmt.Errorf("`query` template isn't supported for alert source template")
``` 
The error messages from the past and present are basically the same.
```
before: can not exec source template errors(1): key "tpl", template "xxx": error evaluating annotation template: template: :1:330: executing "" at <query "sort_desc(process_resident_memory_bytes)">: error calling query: `query` template isn't supported for alert source template

now: cannot template alert source: errors(1): key "tpl", template "xxxx": error evaluating annotation template: template: :1:330: executing "" at <query "sort_desc(process_resident_memory_bytes)">: error calling query: cannot execute query "sort_desc(process_resident_memory_bytes)": query is not available in this context
```

Reply:
error `"`query` template isn't supported in replay mode"` has more context and it's easier to understand for users.

Reply:
ok, reverted, please check